python/lsst/drp/tasks/reprocess_visit_image.py

```diff
@@ -0,0 +1,498 @@
+        ap_corr : `lsst.afw.image.ApCorrMap`
+            Aperture Correction model for this exposure.
+        photo_calib : `lsst.afw.image.PhotoCalib`
+            Photometric calibration model for this exposure.
+        wcs : `lsst.afw.geom.SkyWcs`
+            World Coordinate System model for this exposure.
+        summary_stats : `lsst.afw.image.ExposureSummaryStats`
+            Summary statistics measured on this exposure.
+        id_generator : `lsst.meas.base.IdGenerator`, optional
+            Object that generates source IDs and provides random seeds.
+        result : `lsst.pipe.base.Struct`, optional
+            Result struct that is modified to allow saving of partial outputs
+            for some failure conditions. If the task completes successfully,
+            this is also returned.
+
+        Returns
+        -------
+        result : `lsst.pipe.base.Struct`
+            Results as a struct with attributes:
+
+            ``exposure``
+                Calibrated exposure, with pixels in nJy units.
+                (`lsst.afw.image.Exposure`)
+            ``source``
+                Sources that were measured on the exposure calibrated fluxes
+                and magnitudes.
+                (`astropy.table.Table`)
+            ``source_footprints``
+                Footprints of sources that were measured on the exposure.
+                (`lsst.afw.table.SourceCatalog`)
+        """
+        if result is None:
+            result = pipeBase.Struct()
+        if id_generator is None:
+            id_generator = lsst.meas.base.IdGenerator()
+
+        result.exposure = self.snap_combine.run(exposures).exposure
+
+        # Calibrate the image, so it's on the same units as the background.
+        result.exposure.maskedImage = photo_calib.calibrateImage(result.exposure.maskedImage)
+        result.exposure.maskedImage -= background.getImage()
+        # Uncalibrate so that we do the measurements in instFlux, because
+        # we don't have a way to identify measurements as being in nJy (TODO!).
+        result.exposure.maskedImage /= photo_calib.getCalibrationMean()
+
+        result.exposure.setPsf(psf)
+        result.exposure.setApCorrMap(ap_corr)
+        result.exposure.setWcs(wcs)
+        result.exposure.info.setSummaryStats(summary_stats)
+        # Note: we don't set photoCalib here, because we have to use it below
+        # to calibrate the image and catalog, and thus the image will have a
+        # photoCalib of exactly 1.
+
+        result.source_footprints = self._find_sources(result.exposure, background, id_generator)
+        result.source_footprints["visit"] = result.exposure.visitInfo.id
+        result.source_footprints["detector"] = result.exposure.info.getDetector().getId()
+        result.background = background
+        self._apply_photo_calib(result.exposure, result.source_footprints, photo_calib)
+
+        # TODO: do we need this, or can we drop LocalWcs/LocalPhotoCalib?
```

Comment:
The LocalPhotoCalib will be 1.0 everywhere I guess.  But the LocalWcs still is a vector field that helps a catalog user know about the state of the WCS at the position of a source.  (We aren't warping).

Reply:
This is another bare TODO